### PR TITLE
Don't term_to_binary as much when serializing

### PR DESCRIPTION
**Pull request by Vagabond**
Relates to helium/relcast#50

### CODE REVIEW

File: src/hbbft_acs.erl [hunk 1]
```diff
@@ -257,8 +257,7 @@ sort_bba_msgs(Msgs) ->
 -spec serialize(acs_data()) -> #{atom() => #{} | binary()}.
 serialize(#acs_data{done=Done, n=N, f=F, j=J, rbc=RBCMap, bba=BBAMap}) ->
     M = #{done => Done, n => N, f => F, j => J},
-    M1 = maps:map(fun(_K, V) -> term_to_binary(V, [compressed]) end, M),
-    M1#{rbc => serialize_state(RBCMap, rbc),
+    M#{rbc => serialize_state(RBCMap, rbc),
```
Comment (marcsugiyama) [inline src/hbbft_acs.erl:260]: Could be in a single map construction, although maybe the compiler optimizes to that?

File: src/hbbft_acs.erl [hunk 2]
```diff
@@ -269,7 +268,12 @@ deserialize(#acs_serialized_data{done=Done, n=N, f=F, j=J, rbc=RBCMap, bba=BBAMa
 deserialize(M, SK) ->
     {RBCMap, M1} = maps:take(rbc, M),
     {BBAMap, M2} = maps:take(bba, M1),
-    M3 = maps:map(fun(_K, V) -> binary_to_term(V) end, M2),
+    M3 = maps:map(fun(_K, V) when is_binary(V) ->
```
Comment (marcsugiyama) [inline src/hbbft_acs.erl:271]: Is maps:filter clearer? Can also use a function head match instead of a guard for `is_binary`:```
fun(_K, <<V/binary>>) -> ...
```

File: src/hbbft_rbc.erl [hunk 3]
```diff
@@ -280,15 +280,15 @@ hash_key(Map) ->
               end, #{}, Map).
 
 serialize(#rbc_data{stripes=Stripes}=Data) when Stripes /= #{} ->
-    #{rbc_data => term_to_binary(Data#rbc_data{stripes=#{}}), stripes =>
+    #{rbc_data => Data#rbc_data{stripes=#{}}, stripes =>
```
Comment (marcsugiyama) [inline src/hbbft_rbc.erl:283]: The formatting here makes this hard to follow, maybe put each key on a separate line?
```
#{
  rbc_data => Data#rbc_data{strips = #{}},
  stripes =>
    maps:map( ... )
}
```

**Review by marcsugiyama — COMMENTED** [review]
`hbbt:serialize_hbbft_data` calls `term_to_binary` with `compressed` when serializing the `hbbtf_data`. With the changes to `relcast`, can this instead leave the fields in the original form?

File: src/hbbft_acs.erl (hunk 2 above)
Comment (Vagabond) [reply src/hbbft_acs.erl:271]: I thnk this code is unnecessary entirely, will remove.

**Comment by Vagabond** [comment]
> `hbbt:serialize_hbbft_data` calls `term_to_binary` with `compressed` when serializing the `hbbtf_data`. With the changes to `relcast`, can this instead leave the fields in the original form?

yep!

**Review by marcsugiyama — APPROVED** [review]

**Review by evanmcc — APPROVED** [review]
